api/src/test/java/io/strimzi/api/kafka/model/nodepool/KafkaNodePoolCrdIT.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.nodepool;
+
+import io.strimzi.api.kafka.model.AbstractCrdIT;
+import io.strimzi.test.TestUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * This class tests the encoding and decoding of the KafkaNodePool custom resource
+ */
+public class KafkaNodePoolCrdIT extends AbstractCrdIT {
```

Comment:
should we also have a test where the `KafkaNodePool` resource has a wrong/not existing role configured in?